File: src/main/java/ru/nucodelabs/gem/view/control/VBUserControl.java
package ru.nucodelabs.gem.view.control;

import javafx.scene.layout.VBox;
import ru.nucodelabs.gem.util.fx.FXUtils;

/**
 * Abstract class of user control which have VBox as root container.
 * You must create same named FXML-file in same package in resources folder.
 */
public abstract class VBUserControl extends VBox {

    public VBUserControl() {
        FXUtils.initFXMLControl(this);
    }
}

File: src/main/java/ru/nucodelabs/gem/view/control/VBUserControl.kt
package ru.nucodelabs.gem.view.control

import javafx.scene.layout.VBox
import ru.nucodelabs.gem.util.fx.FXUtils

/**
 * Abstract user control which has VBox as a root container.
 * The FXML file must have the same name and be located in the same package.
 */
abstract class VBUserControl : VBox() {
    init {
        FXUtils.initFXMLControl(this)
    }
}
